intermediate-exercises/src/components/01-LifeCycle.js

```diff
@@ -11,34 +11,37 @@ import React from 'react';
 class LifeCycle extends React.Component {
   constructor(props) {
     super(props);
+
     this.state = { name: 'Pesto' };
+
+    this.onClick = this.onClick.bind(this);
   }
 
-  // This code will be called when the component finishes mounting
-  // (so it is visible for a user).
   componentDidMount() {
-    // Task 1: Display a message "I'm mounted!" in developer's console when the
-    //         component finishes mounting.
-    //         Use `console.log` function for it.
+    console.log("I'm mounted!");
   }
 
   componentDidUpdate(prevProps, prevState) {
-    // Task 2: Display a message "Updated!" in developer's console
-    //         when the component updates.
-    //         Here you also need to use the console.log function.
-    //         Notice that in this lifecycle method you have an access
-    //         to previous values of properties and state.
+    console.log('Updated!');
     console.log({ prevProps, prevState });
   }
 
   componentWillUnmount() {
-    // Task 3: Display a message "Good night, Pesto!" in developer's
-    //         console when the component unmounts.
+    console.log('Good night, Pesto!');
+  }
+
+  onClick(event) {
+    this.setState({ name: 'India' });
   }
 
   // Task 4: Create a button which triggers ComponentDidUpdate lifecycle method on click.
```

Comment:
@rakeshtembhurne , please do delete these comments once you're done with the question.

Reply:
Oh yes, they will be removed in the next commit.